pkg/transforms/policy.go

```diff
@@ -37,6 +41,121 @@ func PolicyResourceBuilder(p *p.Policy) *PolicyResource {
+func getPolicyCommonProperties(c *unstructured.Unstructured, node Node) Node {
+	node.Properties["annotations"] = c.GetAnnotations()
+	node.Properties["isExternal"] = false
+
+	for _, m := range c.GetManagedFields() {
+		if m.Manager == "multicluster-operators-subscription" ||
+			strings.Contains(m.Manager, "argocd") {
+			node.Properties["isExternal"] = true
+		}
+	}
+
+	typeMeta := metav1.TypeMeta{
+		Kind:       c.GetKind(),
+		APIVersion: c.GetAPIVersion(),
+	}
+
+	apiGroupVersion(typeMeta, &node) // add kind, apigroup and version
+
+	policyNsName, ok := c.GetLabels()["policy.open-cluster-management.io/policy"]
+	if ok {
+		stringSlice := strings.Split(policyNsName, ".")
+		// If len(stringSlice) != 2, failed to parse policyNsName
+		if len(stringSlice) == 2 {
+			node.Properties["parentPolicyNs"] = stringSlice[0]
+			node.Properties["parentPolicyName"] = stringSlice[1]
+		}
+	}
+
+	complianceState, found, err := unstructured.NestedString(c.Object, "status", "compliant")
+	if !found || err != nil {
+		node.Properties["compliant"] = ""
+	} else {
+		node.Properties["compliant"] = complianceState
+	}
+
+	severity, found, err := unstructured.NestedString(c.Object, "spec", "severity")
+	if !found || err != nil {
+		node.Properties["severity"] = ""
+	} else {
+		node.Properties["severity"] = severity
+	}
+
+	// remediationAction is same as `remediationAction`
+	remediationAction, found, err := unstructured.NestedString(c.Object, "spec", "remediationAction")
+	if !found || err != nil {
+		node.Properties["remediationAction"] = ""
+	} else {
+		node.Properties["remediationAction"] = remediationAction
+	}
+
+	return node
+}
+
+// For cert, config policies.
+func StandalonePolicyResourceBuilder(c *unstructured.Unstructured) *PolicyResource {
+	node := transformCommon(c) // Start off with the common properties
+
+	return &PolicyResource{node: getPolicyCommonProperties(c, node)}
+}
+
+func OperatorPolicyResourceBuilder(c *unstructured.Unstructured) *PolicyResource {
+	node := transformCommon(c) // Start off with the common properties
+
+	node = getPolicyCommonProperties(c, node)
+
+	var deploymentAvailable string
+	var upgradeAvailable string
+
+	conditions, found, err := unstructured.NestedSlice(c.Object, "status", "conditions")
+	if found && err == nil {
+		for _, condition := range conditions {
+			mapCondition, ok := condition.(map[string]interface{})
+			if !ok {
+				continue
+			}
+
+			conditionType, found, err := unstructured.NestedString(mapCondition, "type")
+			if !found || err != nil {
+				continue
+			}
+
+			conditionReason, found, err := unstructured.NestedString(mapCondition, "reason")
+			if !found || err != nil {
+				continue
+			}
+
+			if conditionType == "InstallPlanCompliant" {
+				if conditionReason == "InstallPlanRequiresApproval" || conditionReason == "InstallPlanApproved" {
+					upgradeAvailable = "yes"
+				} else {
+					upgradeAvailable = "no"
+				}
+			}
+
+			if conditionType == "DeploymentCompliant" {
```

Comment:
This should be `else if`.

Reply:
any `conditionReason` you have in mind?

Reply:
It's a minor optimization since if the previous if statement is true, this second if statement will always be false.